unity-renderer/Assets/UIComponents/Scripts/Components/Carousel/CarouselComponentView.cs

```diff
@@ -565,7 +573,7 @@ internal IEnumerator RunAnimationCoroutine(CarouselDirection direction, int numb
         isInTransition = false;
     }
 
-    internal void GenerateDotsSelector()
+    public void GenerateDotsSelector()
```

Comment:
Why this method has to be `public`? Is it used out of this class?

Reply:
For performance consideration, I wanted to generate Dots only after all Items were added asynchronously in `HighlightsSubSectionComponentView `or in `EventsSubSectionComponentView `(and not after each one was added). Thus, these 2 classes calls this method after getting all Items added

Reply:
a similar idea was behind presenting public `SetItemSizeForModel` in `GridComponentView` - to resize container after all Items were added/set